src/core/qgstemporalnavigationobject.h

```diff
@@ -203,6 +203,14 @@ class CORE_EXPORT QgsTemporalNavigationObject : public QgsTemporalController, pu
      */
     void setLooping( bool loop );
 
+    /**
+     * Finds the best suited framenumber based on the Start of an (old) frame
+     * Purpose is to find the best state of current object if timesteps or extent change.
+     *
+     * @return bestFrameNumber
```

Comment:
```suggestion
```

It's a little too specific, and this function looks generally useful in many cases!